meson.build

```diff
@@ -651,10 +651,18 @@ if get_option('enable-x11')
         ],
     )
 endif
+# TODO: version range?
+icu_dep = dependency('icu-uc', required: false)
+keysyms_test_dep = [test_dep]
+keysyms_test_c_args = ['-DENABLE_PRIVATE_APIS']
+if icu_dep.found()
+    keysyms_test_dep += [icu_dep]
+    keysyms_test_c_args += ['-DHAVE_ICU']
```

Comment:
I think this should be 
```
configh_data.set('HAVE_ICU', 1)
```
and it then gets pulled in via `config.h`

(we also seem to have a mix of `#if` and `#ifdef` for our own constants...)

Reply:
Rebased & fixed.

> (we also seem to have a mix of `#if` and `#ifdef` for our own constants...)
I opted for just `#if`, as it is the most used.